docs/src/developer/reference/oauth.md

```diff
@@ -0,0 +1,402 @@
+### Authorization server
+
+The authorization server does the authentication of the user and establishes whether the user approves or denies the client's access request. In this case the authorization server is the same server as the resource server which is `wire-server`.
+
+## OAuth flow
+
+`wire-server` currently only supports the [authorization code flow](https://www.rfc-editor.org/rfc/rfc6749#section-4.1) which is optimized for confidential clients such as Outlook Calendar Extension.
+
+```{eval-rst}
+.. kroki::
+   :type: mermaid
+   :caption: OAuth 2.0 authorization code flow
+
+   sequenceDiagram
+       autonumber
+       actor U as User
+       participant C as Outlook Calendar Extension
+       participant A as Authorization Server (wire-server)
+       participant R as Resource Server (wire-server)
+
+       U->>C: Click login
+       C->>A: Authorization code request /authorize
+       A->>U: Redirect to login/authorization prompt
+       U->>A: Authenticate and consent
+       A->>C: Authorization code
+       C->>A: Authentication code + client crednetials
+       A->>A: Validate authentication code + client crednetials
+       A->>C: Access token
+       C->>R: Request a resource with access token (e.g. POST /conversations)
+       R->>R: Validate access token with public key from auth server
+       R->>C: Response
+```
+
+### Registering an OAuth client
+
+A new OAuth client can be register via the internal API of `brig` by providing an application name and a redirect URL:
+
+```curl
+  curl -s -X POST localhost:8082/i/oauth/clients \
+    -H "Content-Type: application/json" \
+    -d '{
+      "applicationName":"Outlook Calendar Extension",
+      "redirectUrl":"https://example.com"
+    }'
+```
+
+Parameters:
+
+| Parameter         | Description                                                                                          |
+| ----------------- | ---------------------------------------------------------------------------------------------------- |
+| `redirectUrl`     | The URL to which Wire app will redirect the browser after authorization has been granted by the user |
+| `applicationName` | The name of the application that will be shown on the consent page                                   |
+
+Client credentials will be generated and returned by wire-server:
+
+```json
+{
+  "clientId": "b9e65569-aa61-462d-915d-94c8d6ef17a7",
+  "clientSecret": "3f6fbd62835859b2bac411b2a2a2a54699ec56504ee32099748de3a762d41a2d"
+}
+```
+
+These credentials have to be stored in a safe place and cannot be recovered if they are lost.
+
+### Get authorization code
+
+When the user wants to use the 3rd party app for the first time, they need to authorize it to access Wire resources on their behalf.
+
+To do so the user is redirected to a login screen for authentication. Once authenticated, the user can either grant or deny the client's access request and the corresponding scope (a list of permissions to give to the 3rd party app).
+
```

Comment:
A lot of what comes below is mostly taken from [RFC6749](https://www.ietf.org/rfc/rfc6749.txt).

I'd feel much more comfortable if we'd simply refer to that, instead of duplicating it here, so people don't need to put both side by side to check if there's any differences.

Reply:
Do you still plan to address this?

Reply:
Partly, yes. I discussed with @fisx that it makes sense to have a concise reference for OAuth client developers, so that they do not have to skim through the noise of the complete RFC.
I am going to reference the RFC and additionally provide a documentation of the subset that we implemented.
This is also quite helpful for anyone new to the topic to get an overview and basic understand of what this is about.